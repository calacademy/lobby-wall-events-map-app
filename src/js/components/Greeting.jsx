'use strict'

var React = require('react')

var Greeting = React.createClass({

  displayName: "Greeting",

  getInitialState: function() {
    return {
      items: [
        'We speak your language',
        '我们可以说您的母语。',
        'Nosotros hablamos su idioma.',
        'Nakapagsasalita kami ng Wika mo',
        'お客様の言語で提供いたします'
      ],
      activeItem: 0,
    }
  },

  _nextItem: function() {
      var item = this.state.activeItem + 1 < this.state.items.length ? this.state.activeItem + 1 : 0
      this.setState({activeItem: item})
  },

  componentDidMount: function() {
    setInterval(()=>this._nextItem(), 3000)
  },

  render: function() {

    var _this = this
    var items = this.state.items
    var item = items.map(function(item, index) {
      return (
        <div className="greeting-item notranslate" data-active={index === _this.state.activeItem} key={index}>{item}</div>
      )
    })

    return (
      <div className="greeting">
        <div className="greeting-container">{item}</div>
      </div>
    )

  }
})

export default Greeting
