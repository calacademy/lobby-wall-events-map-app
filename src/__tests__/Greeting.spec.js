import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Greeting from '../js/components/Greeting'

describe('Greeting', () => {

  let greeting

  beforeEach(() => {
    greeting = shallow(<Greeting />)
  })

  it('Class of rendered outer div', () => {
    expect(greeting.find('.greeting').length).toEqual(1)
  })

  it('Class of rendered container div', () => {
    expect(greeting.find('.greeting-container').length).toEqual(1)
  })

  it('Class of rendered iterative divs', () => {
    expect(greeting.find('.greeting-item').length).toEqual(5)
  })

  it('Greeting item divs render data', () => {
    const arr = greeting.find('.greeting-item')
    expect(arr.at(0).text()).toEqual('We speak your language')
    expect(arr.at(1).text()).toEqual('我们可以说您的母语。')
    expect(arr.at(2).text()).toEqual('Nosotros hablamos su idioma.')
    expect(arr.at(3).text()).toEqual('Nakapagsasalita kami ng Wika mo')
    expect(arr.at(4).text()).toEqual('お客様の言語で提供いたします')
  })

  // regression test - jest snapshot
  it('enzyme shallow matches snapshot', () => {
    expect(shallowToJson(greeting)).toMatchSnapshot()
  })

})
