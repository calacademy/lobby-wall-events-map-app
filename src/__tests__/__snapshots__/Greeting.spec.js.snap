exports[`Greeting enzyme shallow matches snapshot 1`] = `
<div
  className="greeting">
  <div
    className="greeting-container">
    <div
      className="greeting-item notranslate"
      data-active={true}>
      We speak your language
    </div>
    <div
      className="greeting-item notranslate"
      data-active={false}>
      我们可以说您的母语。
    </div>
    <div
      className="greeting-item notranslate"
      data-active={false}>
      Nosotros hablamos su idioma.
    </div>
    <div
      className="greeting-item notranslate"
      data-active={false}>
      Nakapagsasalita kami ng Wika mo
    </div>
    <div
      className="greeting-item notranslate"
      data-active={false}>
      お客様の言語で提供いたします
    </div>
  </div>
</div>
`;
